jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/deploy/SamDeployDialog.kt

```diff
@@ -209,20 +194,10 @@ class SamDeployDialog(
         throw error
     }
 
-    private fun createBaseCommand(): CompletionStage<GeneralCommandLine> {
-        val envVars = mutableMapOf<String, String>()
-        envVars.putAll(region.toEnvironmentVariables())
-        envVars.putAll(credentialsProvider.resolveCredentials().toEnvironmentVariables())
-
-        return ExecutableManager.getInstance().getExecutable<SamExecutable>().thenApply {
-            val samExecutable = when (it) {
-                is ExecutableInstance.Executable -> it
-                else -> throw RuntimeException((it as? ExecutableInstance.BadExecutable)?.validationError)
-            }
-            return@thenApply samExecutable
-                .getCommandLine()
-                .withWorkDirectory(template.parent.path)
-                .withEnvironment(envVars)
+    private fun createBaseCommand(): CompletionStage<GeneralCommandLine> = ExecutableManager.getInstance().getExecutable<SamExecutable>().thenApply {
+        when (it) {
+            is ExecutableInstance.Executable -> return@thenApply it.getCommandLine()
```

Comment:
do you even need the `return` here? I'm assuming it should use the last evaluated expression.

Reply:
Cleaned it up, its gone now